src/calendar/calendar.tsx

```diff
@@ -0,0 +1,394 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import dayjs, { Dayjs } from "dayjs";
+import {
+    ArrowLeft,
+    ArrowRight,
+    CalendarDaySection,
+    Circle,
+    Container,
+    DayNumber,
+    DayView,
+    DropdownMonth,
+    DropdownYear,
+    GrowDayCell,
+    HeaderDropdown,
+    InteractiveCircle,
+    LeftCell,
+    MonthView,
+    RightCell,
+    Views,
+    WeekCell,
+    WeekDaysContainer,
+    YearView,
+} from "./calendar.style";
+import { Text } from "src/text";
+import { Icon } from "src/icon";
+import { CalendarProps, VariantDay, View } from "./types";
+import { CalendarMonth } from "./calendar-month";
+import { CalendarYear } from "./calendar-year";
+
+export const Calendar = ({ disabledDate, onChange, value }: CalendarProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [calendarDate, setCalendarDate] = useState<Dayjs>(dayjs());
+    const [showView, setShowView] = useState<View>("Day");
+    const [selectedStartDate, setSelectedStartDate] = useState<string>(""); // YYYY-MM-DD
+
+    useEffect(() => {
+        if (!value) return;
+
+        setSelectedStartDate(value);
+    }, [value]);
+
+    // =============================================================================
+    // Helper Functions
+    // =============================================================================
+
+    const revertedDateFormat = (date: string) => {
+        // DD-MM-YYYY to YYYY-MM-DD
+        return date.split("-").reverse().join("-");
+    };
+
+    const generateDayClass = (date: string): string => {
+        const classes: string[] = [];
+
+        if (selectedStartDate === date) {
+            classes.push("selected-start");
+        }
+
+        return classes.join(" ");
+    };
+
+    const generateDayStatus = (day: Date) => {
+        const _dateStartWithDay = new Intl.DateTimeFormat("en-GB", {
+            day: "2-digit",
+            month: "2-digit",
+            year: "numeric",
+        })
+            .format(day)
+            .replaceAll("/", "-");
+
+        // YYYY-MM-DD
+        const dateStartWithYear = revertedDateFormat(_dateStartWithDay);
```

Comment:
refer to prev comment about converting `Dayjs` to `Date`. can use `dayjs.format()` directly

Reply:
Yes. Both way are workable.

Reply:
if there's no other use for `Date` other than formatting, I think we should stick to `Dayjs` within the component. then no need to convert back and forth

Reply:
My bad. check it again for the reason of using toDate().

Native Date string:  Thu Mar 02 2023 00:00:00 GMT+0800 (Malaysia Time). ----->  toDate()
Date string: dayjs().format("YYYY-MM-DD") or dayjs().format(). That's a string, not Date String. 

https://github.com/LifeSG/react-design-system/blob/3080b00782e8e1930a4212a2791803ed3223c723/src/date-picker/date-picker.tsx#L351

As this example the day will to handle in other way instead using the Native Browser Fn.
```
 calendarDate?.clone().toDate().getMonth() !== day.getMonth()
```